src/Data/Vec/Relation/Binary/Reasoning/Propositional.agda

```diff
@@ -0,0 +1,139 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- The basic code for equational reasoning with displayed propositional equality over vectors
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+module Data.Vec.Relation.Binary.Reasoning.Propositional {a} {A : Set a} where
+
+open import Data.Nat.Base using (ℕ; zero; suc)
+open import Data.Nat.Properties using (suc-injective)
+open import Data.Vec.Base
+open import Relation.Binary.Core using (REL; _⇒_)
+open import Relation.Binary.Definitions using (Sym; Trans)
+open import Relation.Binary.PropositionalEquality.Core
+  using (_≡_; refl; trans; sym; cong; module ≡-Reasoning)
+
+private
+  variable
+    l m n o : ℕ
+    xs ys zs : Vec A n
+
+-- Duplicate definition of cast-is-id and cast-trans to avoid circular dependency
```

Comment:
What is the circular dependency here?

If you want to import `Data.Vec.Properties`, but then use this reasoning module for subsequent lemmas in the rest of that file, then the 'obvious', but perhaps painful, solution, is to make the contents of your `Data.Vec.Relation.Binary.Reasoning.Propositional` a *sub*module of `Data.Vec.Properties`. 

The alternative is to make the parametrisation of this module more complex, and then re-import it in `Data.Vec.Properties` once the `cast` lemmas have been proved. Compare [how this is done in `Data.Integer.Properties`](https://github.com/agda/agda-stdlib/blob/fd541d090bd2cb18f7eedd4c74d33fff2f953828/src/Data/Integer/Properties.agda#L361-L375) for example


Suggest we poll @MatthewDaggitt on the best way to proceed here.

Reply:
I suggest we simply re-export this version of the proofs in `Data.Vec.Properties`

Reply:
Done!

To avoid name collisions, I keep `CastReasoning.cast-is-id` & `CastReasoning.cast-trans` private and re-implemented `Data.Vec.Properties.cast-{is-id,trans}` in terms of `≈-reflexive` & `≈-trans`.